### DISCUSSION TITLE
[#995492] Dedicated server crashes with an Editor project

### DISCUSSION THREAD

**Discussion by TwistedAsylumMC** [Mojang/minecraft-editor#75] [Bugs]
Repro steps
1. Install and run the latest preview build of the dedicated server from https://www.minecraft.net/en-us/download/server/bedrock
2. Create a new project using the Editor client
3. Navigate to `%localappdata\Packages\Microsoft.MinecraftWindowsBeta_8wekyb3d8bbwe\LocalState\games\com.mojang\minecraftWorlds` and copy the new world
4. Replace the default dedicated world with the copied project
5. Start the dedicated server

Observed result vs. what you actually expected to happen
When launching the dedicated server with the editor project, the process terminates without any error messages. Expected behaviour is for the server to launch and the client able to connect and use the editor.

Other helpful info

    Preview build version: 1.19.80.20 (Build 14327935)
    Dedicated server version: 1.19.80.20
    Frequency: Always
    Device info: Windows 11 22621.1265

**Comment by dacowan** [comment]
@minecraftchloe - can you log this one please, and I'll take a look - I have a feeling that perhaps Editor might not have been properly enabled in the preview release pipeline

**Comment by Hatchibombotar** [comment]
Still crashes on preview 1.20.0.25

**Reply by dacowan** [reply]
Just to make sure you are following the correct steps.
- Create an Editor project
- navigate to `%LOCALAPPDATA%\Packages\Microsoft.MinecraftWindowsBeta_8wekyb3d8bbwe\LocalState\games\com.mojang\minecraftWorlds`
- Copy the world folder (e.g. `4kdlZKKaAAA=`) to `<BedrockServerInstallFolder>\worlds`
- **IMPORTANT** - Rename the folder `4kdlZKKaAAA=` to something else (remove the ending equals sign)
- **IMPORTANT** - Open `<BedrockServerInstallFolder>\server.properties` and locate the key `level-name` and set it to whatever you've renamed the world folder to (e.g. `level-name=4kdlZKKaAAA`)

Now start dedicated_server.exe

**Reply by Hatchibombotar** [reply]
Yep, those are the steps I've taken. It works fine with regular worlds.
I've also tried the following:
- Adding `editor` and `server_editor_library` packs (from the WindowsApps directory) to the behavior_packs folder alongside the other vanilla packs
- Removing `from_disk` from the pack references in `valid_known_packs.json`
- Adding `@minecraft/server-editor-bindings` to the permissions.json file

From Hexland in the Bedrock Addons Discord:
> Only works on Win32 BDS for now
[Link to message](https://discord.com/channels/523663022053392405/1084090299120373760/1084132748786143296)

Could this be why its happening? If so, where would I go to download the win32 version of BDS

Thanks for the support!

**Reply by dacowan** [reply]
You can download the Bedrock Dedicated Server from here -- https://www.minecraft.net/en-us/download/server/bedrock

The version you want is `MINECRAFT DEDICATED SERVER SOFTWARE FOR WINDOWS PREVIEW`

**Reply by Hatchibombotar** [reply]
Yep, that is the one I was downloading.

**Reply by Hatchibombotar** [reply]
Does not work in [preview 1.20.10.23](https://minecraft.azureedge.net/bin-win-preview/bedrock-server-1.20.10.23.zip)

**Reply by dacowan** [reply]
Would you mind zipping a simple world and properties file that you're having problems with, and I'll take a look.  I am unable to repo the issue with files here.

**Reply by Hatchibombotar** [reply]
These are the two
[My Project.zip](https://github.com/Mojang/minecraft-editor/files/11749889/My.Project.zip)
[server.properties.txt](https://github.com/Mojang/minecraft-editor/files/11749910/server.properties.txt)

(server.properties is named correctly, just renamed so it could be uploaded)

**Reply by dacowan** [reply]
I'm sorry - but I can't repo your crash with my current build and your world/properties.

I executed dedicated server once to let it get set up.
I then compared the `server.properties` file with the one you sent (both are identical)
I then copied the contents of your ZIP file into the `worlds/Bedrock Level` folder
> **note**: server.properties says that the world to load is `Bedrock Level` but the level_name in your zip file is called `My Project` - generally, this *shouldn't* be a problem because the `server.properties::level-name` field is only used for the world folder within the 'worlds' folder -- but I just wanted to bring it to your attention.
I executed Bedrock Dedicated Server Preview and got the following output
```
[2023-06-16 12:00:17:037 INFO] Starting Server
[2023-06-16 12:00:17:039 INFO] Version: 1.20.20-preview0
[2023-06-16 12:00:17:041 INFO] Session ID: 9e05a5c3-d075-48ac-917e-ff7a598e3b8a
[2023-06-16 12:00:17:043 INFO] Build ID: development
[2023-06-16 12:00:17:045 INFO] Branch: dacowan/sample-extraction-for-starter-kit
[2023-06-16 12:00:17:048 INFO] Commit ID: 668e03c961204898966b3b4f43687c4b6db6a3a2
[2023-06-16 12:00:17:051 INFO] Configuration: Release
[2023-06-16 12:00:17:059 INFO] Level Name: Bedrock level
[2023-06-16 12:00:17:129 INFO] Game mode: 0 Survival
[2023-06-16 12:00:17:130 INFO] Difficulty: 1 EASY
[2023-06-16 12:00:17:132 INFO] Content logging to console is enabled.
[2023-06-16 12:00:17:133 INFO] set session id to platform from minecraft eventing: ab5bf5e2-a594-4686-8cc4-c16304d0d874
[2023-06-16 12:00:17:326 INFO] Enabling Editor Services
[2023-06-16 12:00:17:328 INFO] Adding world packs from this directory: worlds/Bedrock level
[2023-06-16 12:00:17:659 INFO] Content logging to disk is enabled. Writing log to: ContentLog__Friday__2023_June_16__12_00_17
[2023-06-16 12:00:17:664 INFO] Level Storage logging enabled. Writing log to: C:/Users/dacowan/AppData/Roaming/logs/LevelStoragePerfLog
[2023-06-16 12:00:17:667 INFO] Opening level 'worlds/Bedrock level/db'
[2023-06-16 12:00:17:707 INFO] LevelDB worlds/Bedrock level/db status: OK
```

I then launched the Preview Edition of Minecraft and attached to the local server at 127.0.0.1

<img width="899" alt="image" src="https://github.com/Mojang/minecraft-editor/assets/37003198/fa4ed7a9-c583-4a1b-ae93-8212ba263ecb">

Everything seems to be working fine.

Maybe you could check your log file and see if there are any obvious errors?

**Reply by Hatchibombotar** [reply]
This is my console output:
```
NO LOG FILE! - setting up server logging...
[2023-06-19 15:42:22:708 INFO] Starting Server
[2023-06-19 15:42:22:714 INFO] Version: 1.20.10-beta23
[2023-06-19 15:42:22:717 INFO] Session ID: 75cce044-0a14-4967-9c27-68ca5fb5ab9d
[2023-06-19 15:42:22:719 INFO] Build ID: 16006386
[2023-06-19 15:42:22:721 INFO] Branch: r/20_u1
[2023-06-19 15:42:22:722 INFO] Commit ID: d3f199115ec29dfa3b2a8b1a8c75ac8acd92b9f0
[2023-06-19 15:42:22:725 INFO] Configuration: Publish
[2023-06-19 15:42:22:727 INFO] Level Name: Bedrock level
[2023-06-19 15:42:22:732 INFO] Game mode: 0 Survival
[2023-06-19 15:42:22:734 INFO] Difficulty: 1 EASY
[2023-06-19 15:42:22:736 INFO] Content logging to console is enabled.
```
From google and [a jira bug report](https://bugs.mojang.com/browse/BDS-56), it seems that the log file hasn't worked for at least 5 years - at least on the public versions of bds. I couldn't find a content log where ones from the preview or release builds of the game would be, nor in the location that your console output said Level Storage logging may be found. 

Checking the windows event viewer, it logs an error:
```
Faulting application name: bedrock_server.exe, version: 0.0.0.0, time stamp: 0x6487ad5d
Faulting module name: bedrock_server.exe, version: 0.0.0.0, time stamp: 0x6487ad5d
Exception code: 0xc0000005
Fault offset: 0x0000000000814aeb
Faulting process ID: 0x0x4BBC
Faulting application start time: 0x0x1D9A2BCAD2475F2
Faulting application path: C:\Users\Hatchi\Documents\bedrock-server-1.20.10.23\bedrock_server.exe
Faulting module path: C:\Users\Hatchi\Documents\bedrock-server-1.20.10.23\bedrock_server.exe
Report ID: 92a856fe-82b2-47e2-80b3-a4d921efa9aa
Faulting package full name: 
Faulting package-relative application ID: 
```

Alos one log of type information
```
Fault bucket 1620008052986713209, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: bedrock_server.exe
P2: 0.0.0.0
P3: 6487ad5d
P4: bedrock_server.exe
P5: 0.0.0.0
P6: 6487ad5d
P7: c0000005
P8: 0000000000814aeb
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.1fb9b344-9ce4-4fd8-8dcb-3542773be8ec.tmp.WERInternalMetadata.xml

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_bedrock_server.e_60e29ade454d514b7794be415b7ce5426378e9_f53a8549_ce08b9aa-3d0f-4781-81cd-99b513e1dccf

Analysis symbol: 
Rechecking for solution: 0
Report Id: 92a856fe-82b2-47e2-80b3-a4d921efa9aa
Report Status: 268435456
Hashed bucket: 53bbdfe492a221b8767b6cbdb38b9479
Cab Guid: 0
```

Also The report from C:\ProgramData\Microsoft\Windows\WER\ReportArchive:
[Report.wer.txt](https://github.com/Mojang/minecraft-editor/files/11789355/Report.wer.txt)

Is it possible that editor mode is not enabled on bds, like how it is not currently enabled in stable releases? It doesn't seem very clear what the issue is from the small amount of what I could find.

**Reply by dacowan** [reply]
Looks like there's a bug in the build pipeline for this variant.  I'll try and get the fix in for the next preview push - if it doesn't go in there, it will definitely be fixed for the next one.
I'll let you know when I've confirmed that the published preview build has been successfully updated

**Reply by dacowan** [reply]
Looks like the changes made it through in time for today's new preview build (1.20.10.24)-- would you mind grabbing the new version of BDS -- 

https://minecraft.azureedge.net/bin-win-preview/bedrock-server-1.20.10.24.zip
